lib/item/item.py

```diff
@@ -2018,17 +2018,38 @@ def _log_build_text(self, value, caller, source=None, dest=None):
         mlvalue = self._log_mapping.get(lvalue, lvalue)
         name = self._name
         age = round(self._get_last_change_age(), 2)
-        pname = self.__parent._name
+        try:
```

Comment:
Wozu? _name kann nicht undefiniert sein. _name wird zumindest mit _path initialisiert.

Reply:
Ich hatte bei mir mehrfach diesen Error:
```
2023-11-02  11:41:33 ERROR    cherrypy.error.4478625968 [02/Nov/2023:11:41:33] HTTP 
Traceback (most recent call last):
  File "/Users/ank/Library/Python/3.9/lib/python/site-packages/cherrypy/_cprequest.py", line 638, in respond
    self._do_respond(path_info)
  File "/Users/ank/Library/Python/3.9/lib/python/site-packages/cherrypy/_cprequest.py", line 697, in _do_respond
    response.body = self.handler()
  File "/Users/ank/Library/Python/3.9/lib/python/site-packages/cherrypy/lib/encoding.py", line 223, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/Users/ank/Library/Python/3.9/lib/python/site-packages/cherrypy/_cpdispatch.py", line 54, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/Users/ank/Documents/Github/smarthome/modules/admin/itemdata.py", line 346, in item_change_value_html
    item(value, caller='admin')
  File "/Users/ank/Documents/Github/smarthome/lib/item/item.py", line 1380, in __call__
    self.__update(value, caller, source, dest, key, index)
  File "/Users/ank/Documents/Github/smarthome/lib/item/item.py", line 2232, in __update
    self._set_value(value, caller, source, dest, prev_change=None, last_change=None)
  File "/Users/ank/Documents/Github/smarthome/lib/item/item.py", line 2188, in _set_value
    self._log_on_change(value, caller, source, dest)
  File "/Users/ank/Documents/Github/smarthome/lib/item/item.py", line 2127, in _log_on_change
    txt = self._log_build_text(value, caller, source, dest)
  File "/Users/ank/Documents/Github/smarthome/lib/item/item.py", line 2021, in _log_build_text
    pname = self.__parent._name
AttributeError: 'Items' object has no attribute '_name'
```

Reply:
Das Problem muss aber wo anders liegen. **_name** wird in der Init Routine des Items initialisiert (Zeile 184), kann also nicht nicht vorhanden sein.

Der Fehler besagt auch etwas anderes, nämlich dass das das Object **Items** (nicht das Objekt **Item**) kein Attribut **_name** hat. **items** Ist das Objekt, welches alle Items enthält. Dieses Objekt hat nie ein Atttibut **_name**.

Hat bei Dir das Item bei dem das Problem auftritt evtl. keinen Parent? Das sollte eigentlich nur für das Wurzel Element des Item-Trees auftreten.

Reply:
Jup ist ein root item.

Reply:
Dann wäre es sinnvoller/verständlicher zu schauen welchen Wert **__parent** dann hat und das abzufangen, statt auf **_name** zu prüfen.

Reply:
Dann wäre dein Vorschlag mehr in die Richtung..?
```
if self.__parent = _items_instance:
    pname = None
    pid = None
else:
    pname = self.__parent._name
    pid = self.__parent._path
```
Sonst gerne konkreten Vorschlag posten

Reply:
Ja

Reply:
OK, ist nun drin